src/wl_window.c

```diff
@@ -499,13 +528,22 @@ void _glfwPlatformMaximizeWindow(_GLFWwindow* window)
 
 void _glfwPlatformShowWindow(_GLFWwindow* window)
 {
-    wl_shell_surface_set_toplevel(window->wl.shell_surface);
+    if (!window->monitor)
```

Comment:
Documentation for `glfwHideWindow()` says “If the window is already hidden or is in full screen mode, this function does nothing.” which means it shouldn’t be possible for a window to be both fullscreen (`window->monitor != NULL`) and hidden (`window->wl.visible == GLFW_FALSE`).